### PR TITLE
Propose to add new default cluster-role for storage-admin

### PR DESCRIPTION
**Pull request by screeley44**
storage-admin is someone who is tasked with managing enterprise or project storage needs and assets but not necessarily involved in the management of the cluster or the development or maintenance of the projects/applications.  This person essentially owns/admins some level of storage in the enterprise and has been asked to make it available to the cluster and various projects.  This means they will set up StorageClasses, PV, endpoints (gluster), services (gluster, etc...), possibly secrets for the storage/db, petsets, and possibly some basic read ability as well in the cluster.

I've proposed a starting point for this role but would like some review and input as maybe some resources are missing or some are not needed.

### CODE REVIEW

**Comment by screeley44** [comment]
@erinboyd  @eparis @childsb @deads2k   -  we have been talking about this role in our storage review meetings, so thought I'd get the ball rolling, would like some thoughts on this (especially from David wrt if I'm implementing this properly, missing anything, etc...)

File: pkg/cmd/server/bootstrappolicy/constants.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ const (
 	UnauthenticatedGroup    = "system:unauthenticated"
 	ClusterAdminGroup       = "system:cluster-admins"
 	ClusterReaderGroup      = "system:cluster-readers"
+	StorageAdminGroup       = "system:storage-admin"
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/constants.go]: I'm not sure I would auto-bind the role to a non-existent group. I would start by just defining the role

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 2]
```diff
@@ -270,6 +287,7 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 				authorizationapi.NewRule(readWrite...).Groups(buildGroup).Resources("buildlogs").RuleOrDie(),
 				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("resourcequotausages").RuleOrDie(),
 				authorizationapi.NewRule("create").Groups(authzGroup).Resources("resourceaccessreviews", "subjectaccessreviews").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: putting a cluster-scoped resource in the admin role doesn't make sense, since it is intended to be granted within a namespace scope

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 3]
```diff
@@ -213,7 +214,23 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 				authorizationapi.NewRule("create").Groups(buildGroup).Resources(authorizationapi.JenkinsPipelineBuildResource).RuleOrDie(),
 			},
 		},
-
+		{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: StorageAdminRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				authorizationapi.NewRule(readWrite...).Groups(kapiGroup).Resources("bindings", "endpoints", "persistentvolumes", "persistentvolumes/status",
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: I would not expect global read/write access to these things... I would expect this role to grant the pv/pvc/storageclass permissions, then the user be granted an admin role in a particular namespace if they needed to set up secrets or endpoints for their provisioner.

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 4]
```diff
@@ -213,7 +214,23 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
...
+					"secrets", "services").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("events", "limitranges", "namespaces", "namespaces/status",
+					"resourcequotas", "resourcequotas/status", "resourcequotausages").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(projectGroup).Resources("projects").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("persistentvolumeclaims", "persistentvolumeclaims/status").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("securitycontextconstraints", "serviceaccounts").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("pods/binding", "pods/log", "pods/status").RuleOrDie(),
+				authorizationapi.NewRule(readWrite...).Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
+				authorizationapi.NewRule(readWrite...).Groups(appsGroup).Resources("petsets", "petsets/status").RuleOrDie(),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: same here, global petset permissions don't seem right for this role

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 5]
```diff
@@ -213,7 +214,23 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
...
+			},
+			Rules: []authorizationapi.PolicyRule{
+				authorizationapi.NewRule(readWrite...).Groups(kapiGroup).Resources("bindings", "endpoints", "persistentvolumes", "persistentvolumes/status",
+					"secrets", "services").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("events", "limitranges", "namespaces", "namespaces/status",
+					"resourcequotas", "resourcequotas/status", "resourcequotausages").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(projectGroup).Resources("projects").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("persistentvolumeclaims", "persistentvolumeclaims/status").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("securitycontextconstraints", "serviceaccounts").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("pods/binding", "pods/log", "pods/status").RuleOrDie(),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: why does a storage-admin need read access to all pods?

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 6]
```diff
@@ -213,7 +214,23 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
...
+		{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: StorageAdminRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				authorizationapi.NewRule(readWrite...).Groups(kapiGroup).Resources("bindings", "endpoints", "persistentvolumes", "persistentvolumes/status",
+					"secrets", "services").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("events", "limitranges", "namespaces", "namespaces/status",
+					"resourcequotas", "resourcequotas/status", "resourcequotausages").RuleOrDie(),
+				authorizationapi.NewRule(read...).Groups(projectGroup).Resources("projects").RuleOrDie(),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: why does a storage-admin need read access to all projects?

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 7]
```diff
@@ -817,6 +837,17 @@ func GetBootstrapClusterRoleBindings() []authorizationapi.ClusterRoleBinding {
 		},
 		{
 			ObjectMeta: kapi.ObjectMeta{
+				Name: StorageAdminRoleBindingName,
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: I would omit the default rolebinding

File: pkg/cmd/server/bootstrappolicy/policy.go (hunk 3 above)
Comment (screeley44) [inline pkg/cmd/server/bootstrappolicy/policy.go]: @liggitt  - I trimmed down storage admin based on what you mentioned, but I still left the ability for the storage-admin to create endpoints and services based on the idea if I'm a gluster storage-admin, and I'm setting up SC's and PV's for access to my storage, wouldn't I also want to be able to setup the endpoints for gluster as well without having to add an additional step of the cluster-admin giving me access to a particular project to do so?  thoughts?  Also, I left some read perms in for some stuff that was thinking might be useful for a storage-admin to be able to view.

File: pkg/cmd/server/bootstrappolicy/policy.go (hunk 6 above)
Comment (screeley44) [inline pkg/cmd/server/bootstrappolicy/policy.go]: was thinking it might be useful for SC and PV planning or point of reference, but probably not necessary

File: pkg/cmd/server/bootstrappolicy/policy.go (hunk 5 above)
Comment (screeley44) [inline pkg/cmd/server/bootstrappolicy/policy.go]: same as above, but removed and agree, we can start as lean as possible and then extend when needed

**Comment by screeley44** [comment]
@liggitt @erinboyd @eparis @childsb - addressed first round of review comments, might want to trim some more, as we can always extend in the future if we find we need to, but PTAL

File: pkg/cmd/server/bootstrappolicy/policy.go (hunk 3 above)
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: this role will have to be granted at the cluster scope to enable access to PVs. I would not anticipate giving the storage admin control over every endpoint and service in the cluster

**Comment by erinboyd** [comment]
@liggitt Is there was a way to grant admin access to only 'storagey' services and endpoints? I didn't think there was a way around it.

**Comment by liggitt** [comment]
> @liggitt Is there was a way to grant admin access to only 'storagey' services and endpoints? I didn't think there was a way around it. 

Not with a single role. You would grant roles like this:
```
oadm policy add-cluster-role-to-user storage-admin bob
oadm policy add-role-to-user admin bob -n storage-namespace
```

**Comment by screeley44** [comment]
@liggitt - does this look better?  Anything I'm missing or can still remove?

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 8]
```diff
@@ -213,7 +213,16 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 				authorizationapi.NewRule("create").Groups(buildGroup).Resources(authorizationapi.JenkinsPipelineBuildResource).RuleOrDie(),
 			},
 		},
-
+		{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: StorageAdminRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				authorizationapi.NewRule(readWrite...).Groups(kapiGroup).Resources("persistentvolumes", "persistentvolumes/status", "storageclasses").RuleOrDie(),
```
Comment (screeley44) [inline pkg/cmd/server/bootstrappolicy/policy.go]: ah, I had that at one time, must have reverted...let me fix that back up

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 9]
```diff
@@ -213,7 +213,16 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
...
+				authorizationapi.NewRule(read...).Groups(kapiGroup).Resources("events", "limitranges", "namespaces", "namespaces/status",
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: I was expecting to see some sort of access to persistentvolumeclaims. What are you wanting this role to be able to do?

I wasn't expecting the namespace and namespaces/status. What are those for?

Is the read access to limitranges and resourcequotas just for debugging?

Comment (screeley44) [inline pkg/cmd/server/bootstrappolicy/policy.go]: We are expecting this role as someone who doesn't necessarily need to be a developer or fully understand the administration of the cluster (as you stated, if they do have that knowledge and ability they can be promoted to cluster-admin or given admin role on a specific project or even cluster-reader type of role to help assist)  So I'm thinking about this in terms of an enterprise and in the enterprise there are storage-admins, dba's, etc... they control/own storage assets and want to give them the ability to manage/create the storage resources that can be consumed by the cluster with limited knowledge of the cluster.  So jdoe owns 50 nfs servers, OCP cluster needs access to these servers, so cluster-admin gives jdoe a role of storage-admin so he can properly create the storageclasses, pv's ,etc... Also, some of the read perms I'm thinking are things like being able to see quotas and resource limits as they may impact his storage assets, again without being given full admin or other roles.  Namespaces I was thinking might be useful as a reference as he is creating these classes and objects??  Claims I originally had in there, but removed because they again are project scoped and not sure a storage-admin who simply needs to set up classes and pvs would care about pvcs - besides, they can see what pvcs are bound anyway through the PV.  Maybe my thought process on this role is off?  @erinboyd  want to chime in here?

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: I would limit the role to pvs, storageclasses, and maybe pvcs (and maybe readonly? it's not clear to me whether we expect this role to be responsible for resolving PV/PVC issues, which could require write access, or at least read access to PVCs)

Comment (screeley44) [inline pkg/cmd/server/bootstrappolicy/policy.go]: @liggitt  +1 ... will fix it up and as I stated, if we find we need to expand later we can...thanks for looking it over

**Comment by liggitt** [comment]
@deads2k would you expect a `system:` prefix on this one?

**Comment by deads2k** [comment]
> @deads2k would you expect a system: prefix on this one?

No.  We aren't using it for anything internally.  If a user wants to take ownership of it and remove permissions, all of our core components will continue working.

It does seem like running a sample use-case with a basic script to prove that this functionally works might help scope it though.

**Comment by screeley44** [comment]
@deads2k - is there a formal test/script process that is typically used beyond the policy_test.go to help prove functionality?  If so can you point me to an example of a similar test if one exists?  I've tested this manually several times to make sure the policy is working but nothing formal via a script

**Comment by deads2k** [comment]
> @deads2k - is there a formal test/script process that is typically used beyond the policy_test.go to help prove functionality? If so can you point me to an example of a similar test if one exists? I've tested this manually several times to make sure the policy is working but nothing formal via a script

If it only requires the master (no nodes) to see the API changes react, then we commonly use `test-cmd.sh` (well, the scripts under `test/cmd`) to prove basic scripting scenarios.  If it requires more than that, we have `test/extended/cmd.sh`

File: pkg/cmd/server/bootstrappolicy/policy.go (hunk 9 above)
Comment (erinboyd) [inline pkg/cmd/server/bootstrappolicy/policy.go]: @screeley44 @liggitt  I guess it's possible a storage admin would create the PVCs.If we are going to let the storage admin set the quotas, these are project scoped as well, so I don't see what would be the point of leaving pvcs out.

**Comment by screeley44** [comment]
@deads2k @liggitt  -  how does this look? I added a script to test a few scenarios as well

File: pkg/cmd/server/bootstrappolicy/policy.go (hunk 9 above)
Comment (erinboyd) [inline pkg/cmd/server/bootstrappolicy/policy.go]: @screeley44 what 'admin' creates the quotas then if the Storage admin doesn't have access to do this?

Comment (screeley44) [inline pkg/cmd/server/bootstrappolicy/policy.go]: @erinboyd  cluster-admin would have that authority for this role, I'm not sure we want a storage-admin to have that ability to manage cluster-wide resourcequotas, at least that is what I gathered from previous comments above

Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: Quota controls much more than storage. The same person *may* be responsible for both, but isn't necessarily. I'd prefer a separate "quota admin" role over combining unrelated resources in this role

**Comment by screeley44** [comment]
@liggitt  @deads2k - based on PR https://github.com/openshift/origin-web-console/pull/661

should I add _read_ perms for _storageclasses_ to the admin, edit and view roles?  Seems like that would be useful?

**Comment by liggitt** [comment]
no, that won't accomplish anything, since storageclasses are cluster-scoped, and those roles are granted only within particular namespaces. If storageclasses are expected to be visible to all users, adding get/list permissions to the "basic-user" role would be fine (same as we do for clusterroles)

**Comment by screeley44** [comment]
@erinboyd @liggitt @deads2k 

I think this will work if I'm understanding this correctly, by just giving the _list_ perm on basic-user, we are letting them list the available storageclasses but they are not allowed to get or describe them, which will mitigate the risk of users seeing potential confidential information (secrets, endpoints, etc...)  Right?

This is just a normal user who access to 1 project, they can list SCs but can not get any more details on them.

```
[root@ose1 openshift]# oc get storageclasses
NAME       TYPE
generic1   kubernetes.io/gce-pd   
slow       kubernetes.io/gce-pd   

[root@ose1 openshift]# oc get storageclass slow
Error from server: User "dev12" cannot get storage.k8s.io.storageclasses at the cluster scope

[root@ose1 openshift]# oc describe storageclass slow
Error from server: User "dev12" cannot get storage.k8s.io.storageclasses at the cluster scope

```

**Comment by screeley44** [comment]
@eparis  -  PTAL

**Comment by liggitt** [comment]
list is strictly more powerful than get... it includes all the info you would obtain from getting each object individually. storage classes were designed to be seen by end users, so adding get/list/watch to the admin/edit/view roles is fine with me. End users knowing where the secrets for your glusterfs provisioner live would make me unhappy as a storage admin, but oh well.

**Comment by screeley44** [comment]
@liggitt  not sure I'm following, I thought last week we decided the better approach was to add in the basic-user, the ability to see the SC only, since all users get the basic-user role by default.  if I only give "list" to the basic-user role, then all users could basically see the storageclasses when they do a get, but can not see any details, including things like secrets, endpoints, etc... If I give "get" perms as well, then the user has the ability to 'oc get storageclass myclass1' and see all the details of the SC, same with describe...if you notice the output from my tests in the previous message I posted above that shows the user is restricted from describe and _get storageclass <object>_  

Am I missing something?

**Comment by liggitt** [comment]
> Am I missing something?

Nope, I'm losing my mind and forgetting storageclass is cluster-scoped. Yes, adding get/list/watch permission to basic-user makes sense to me

**Comment by screeley44** [comment]
@liggitt cool!  

@eparis  @erinboyd  ... so I think this is ready for a final review pass through

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 10]
```diff
@@ -388,6 +397,7 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 				authorizationapi.NewRule("get").Groups(userGroup).Resources("users").Names("~").RuleOrDie(),
 				authorizationapi.NewRule("list").Groups(projectGroup).Resources("projectrequests").RuleOrDie(),
 				authorizationapi.NewRule("get", "list").Groups(authzGroup).Resources("clusterroles").RuleOrDie(),
+				authorizationapi.NewRule("list").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: surprised unit tests didn't complain about this permission not being in the test fixtures

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 11]
```diff
@@ -625,6 +635,8 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 				// TODO: change glusterfs to use DNS lookup so this isn't needed?
 				// Needed for glusterfs volumes
 				authorizationapi.NewRule("get").Groups(kapiGroup).Resources("endpoints").RuleOrDie(),
+				// Needed for Dynamic Provisioning
+				authorizationapi.NewRule("get").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
```
Comment (liggitt) [inline pkg/cmd/server/bootstrappolicy/policy.go]: surprised unit tests didn't complain about this permission not being in the test fixtures

**Comment by liggitt** [comment]
LGTM, [merge]

**Comment by screeley44** [comment]
@liggitt - any idea what I'm missing in the test fixtures (and which one)?  I think all the updates to policy.go are represented in _bootstrap_cluster_roles.yaml_

**Comment by liggitt** [comment]
run `hack/test-go.sh pkg/cmd/server/bootstrappolicy` and it will tell you what is missing (and if you run that test with an UPDATE_BOOTSTRAP_POLICY_FIXTURE_DATA=1 envvar set, it'll update the fixtures for you)

**Comment by screeley44** [comment]
@liggitt - hmmm

```
[screeley@screeley origin (origin-storage-class-roles)]$ hack/test-go.sh pkg/cmd/server/bootstrappolicy
ok  	github.com/openshift/origin/pkg/cmd/server/bootstrappolicy	1.306s	coverage: 76.0% of statements
hack/test-go.sh succeeded after 54 seconds
```

```
[screeley@screeley origin (origin-storage-class-roles)]$ git status
On branch origin-storage-class-roles
nothing to commit, working directory clean
```

**Comment by screeley44** [comment]
@liggitt -  seems I'm in sync and I can't find a flake, not sure how to get past this, please advise when you get a chance

**Comment by liggitt** [comment]
> @liggitt - seems I'm in sync and I can't find a flake, not sure how to get past this, please advise when you get a chance

when I check out your branch, I see the unit test failure:

```
git checkout https://github.com/openshift/origin/pull/11138
make clean
hack/test-go.sh pkg/cmd/server/bootstrappolicy
	policy_test.go:92: If the change is expected, re-run with UPDATE_BOOTSTRAP_POLICY_FIXTURE_DATA=true to update the fixtures
FAIL
coverage: 76.0% of statements
FAIL	github.com/openshift/origin/pkg/cmd/server/bootstrappolicy	0.343s
[ERROR] PID 16883: hack/test-go.sh:292: `go test ${gotest_flags} ${test_packages}` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/test-go.sh:292: `go test ${gotest_flags} ${test_packages}`
[INFO]   Exiting with code 1.
hack/test-go.sh failed after 73 seconds
```

**Comment by screeley44** [comment]
@liggitt - that is odd, I was finally able to reproduce issue and use the UPDATE to find it, but had to use a fresh VM, my local repo always passed...anyway, it was one space, so see if this makes it through now :-)

**Comment by screeley44** [comment]
@liggitt - can you +1 this again since I updated it?

**Comment by screeley44** [comment]
@openshift-bot test this issue: #11240

**Comment by screeley44** [comment]
#11240 [test]

**Comment by eparis** [comment]
[merge]

**Comment by screeley44** [comment]
@liggitt -  that seems like a flake doesn't it?  It's not anything this PR touches, should I create a new flake for this, or do you notice something else I'm missing?

**Comment by liggitt** [comment]
flaked on #11240, but also failed bootstrap role verification (https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/7349/consoleFull#141260681756bf4006e4b05b79524e5923)

**Comment by liggitt** [comment]
opened https://github.com/screeley44/origin/pull/1 with fixed test data (I just ran `UPDATE_BOOTSTRAP_POLICY_FIXTURE_DATA=true hack/test-go.sh pkg/cmd/server/bootstrappolicy` locally). merge and squash that, then the tests should pass

**Comment by screeley44** [comment]
#11353 [test]

**Comment by screeley44** [comment]
#11057 [test]

**Comment by eparis** [comment]
[merge] is the bot ignoring my commands?!?

**Comment by screeley44** [comment]
@eparis @liggitt - holy cow, it's all green!! Please merge before something changes :-)

**Comment by liggitt** [comment]
You're still in the merge queue, which is limping along :)

File: hack/test-cmd.sh [hunk 12]
```diff
@@ -301,7 +301,7 @@ os::cmd::expect_success 'oc logout'
 os::cmd::expect_failure_and_text 'oc get pods' '"system:anonymous" cannot list pods'
 
 # make sure we handle invalid config file destination
-os::cmd::expect_failure_and_text "oc login '${KUBERNETES_MASTER}' -u test -p test --config=/src --insecure-skip-tls-verify" 'KUBECONFIG is set to a file that cannot be created or modified'
+os::cmd::expect_failure_and_text "oc login '${KUBERNETES_MASTER}' -u test -p test --config=/tmp --insecure-skip-tls-verify" 'Error loading config file'
```
Comment (pmorie) [inline hack/test-cmd.sh]: This looks to me like it should be a separate PR, unless there's something related to the role with this change that I'm somehow missing.

File: pkg/cmd/server/bootstrappolicy/policy.go [hunk 13]
```diff
@@ -217,7 +217,16 @@ func GetBootstrapClusterRoles() []authorizationapi.ClusterRole {
 				authorizationapi.NewRule("create").Groups(buildGroup).Resources(authorizationapi.JenkinsPipelineBuildResource).RuleOrDie(),
 			},
 		},
-
+		{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: StorageAdminRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				authorizationapi.NewRule(readWrite...).Groups(kapiGroup).Resources("persistentvolumes", "persistentvolumes/status").RuleOrDie(),
```
Comment (pmorie) [inline pkg/cmd/server/bootstrappolicy/policy.go]: What about storage-admin requires access to the pv status subresource, and why is read needed at all for the that subresource?

File: pkg/cmd/server/bootstrappolicy/policy.go (hunk 11 above)
Comment (pmorie) [inline pkg/cmd/server/bootstrappolicy/policy.go]: What dynamic provisioning work does the Node role perform that it needs access to this resource?

File: hack/test-cmd.sh (hunk 12 above)
Comment (liggitt) [inline hack/test-cmd.sh]: Agree, please revert this change

**Comment by eparis** [comment]
I openned https://github.com/openshift/origin/pull/11514 to clean up some comments and try to get the merge in. We'll need to pay particular attention to my changes in https://github.com/openshift/origin/pull/11514/commits/e2a78d398f927cf8dd8c30942472ac758fb4c800

**Comment by screeley44** [comment]
@eparis @pmorie - rebased and addressed Paul's comments

**Comment by eparis** [comment]
[test] failure pulling from yum of some sort...

**Comment by eparis** [comment]
[merge]
